src/Transport/Connectivity/MessageSenderLifeCycleManager.cs

```diff
@@ -20,15 +20,13 @@ public IMessageSender Get(string entitypath, string viaEntityPath, string namesp
             var buffer = MessageSenders.GetOrAdd(entitypath + viaEntityPath + namespaceName, s =>
             {
                 var b = new CircularBuffer<EntityClientEntry>(numberOfSendersPerEntity);
-                for (var i = 0; i < numberOfSendersPerEntity; i++)
+                for(var i = 0; i < numberOfSendersPerEntity; i++)
                 {
                     var e = new EntityClientEntry();
-                    lock (e.Mutex)
```

Comment:
In that case, wouldn't you want to 'Get' the next and evict the current? Not sure how expensive it is and how frequent this happens.